realm/realm-library/src/main/java/io/realm/OrderedRealmCollectionImpl.java

```diff
@@ -539,6 +548,7 @@ protected E convertRowToObject(UncheckedRow row) {
         if (className != null) {
             return new OrderedRealmCollectionSnapshot<E>(realm, collection, className);
         } else {
+            assert classSpec != null;
```

Comment:
I added the assertion to suppress warning from IDE.
In this class, if `className` is null, `classSpec` is always non-null, but no way to tell that without assertion.